db/db_impl.cc

```diff
@@ -552,6 +552,10 @@ Status DBImpl::SetDBOptions(
         MaybeScheduleFlushOrCompaction();
       }
 
+      if(new_options.wal_bytes_per_sync !=
+          mutable_db_options_.wal_bytes_per_sync) {
+        Status wal_status = HandleWALFull(&write_context);
+      }
```

Comment:
you also need to update env_options_ after line 564. bytes_per_sync is pass to flush and compaction jobs via env_options_.